vim/core/oni-core-interop/plugin/init.vim

```diff
@@ -42,6 +52,72 @@ function OniNotifyEvent(eventName)
     call OniNotify(["event", a:eventName, context])
 endfunction
 
+function User_buffers() " help buffers are always unlisted, but quickfix buffers are not
+  return filter(range(1,bufnr('$')),'buflisted(v:val) && "quickfix" !=? getbufvar(v:val, "&buftype")')
+endfunction
+
+function OniGetAllBuffers()
+  let l:buffers = []
+  let l:bufnums = User_buffers()
+  if exists("l:bufnums")
+    for l:bufnum in l:bufnums
+      try
+        let l:buffer = OniGetEachContext(l:bufnum)
+        let l:buffers += [l:buffer]
+      catch /.*/
+      "Probably dont want this outside of a debugging scenario
+      " echohl WarningMsg
+      " echo v:exception
+      " echohl none
+    endtry
+    endfor
+    return l:buffers
+  endif
+endfunction
+
+
+function OniGetEachContext(bufnum)
+  let l:context = {}
+    let l:bufpath = bufname(a:bufnum)
+
+    if strlen(l:bufpath)
+      "TODO: python can be used to retrieve the lines a buffer and len return 
+      " the number of files this is much quicker than using viml getbufline fn
+      " https://stackoverflow.com/questions/14544618/is-there-a-way-to-get-the-number-of-lines-of-a-buffer-in-vim-script
+      " if has('python')
+      " python import vim
+      " let l:context.bufferTotalLines = pyeval('len(vim.buffers['.(a:bufnum -1).'])')
+      let l:context.bufferNumber = a:bufnum
+      let l:context.bufferFullPath = expand("#".a:bufnum.":p")
+      let l:context.bufferTotalLines = v:null
+      let l:context.line = v:null
```

Comment:
Do we need all these `v:null`s here? I'm thinking that it might be worth having a different 'type' as opposed to trying to fit this to the eventcontext